browser/lib/util/crypto.js

```diff
@@ -98,84 +115,102 @@ var Crypto = (function() {
 	 * algorithm is the name of the algorithm in the default system provider,
 	 * or the lower-cased version of it; eg "aes" or "AES".
 	 *
-	 * Clients may instance a CipherParams directly and populate it, or may
-	 * query the implementation to obtain a default system CipherParams.
+	 * Clients are recommended to not call this directly, but instead to use the
+	 * Crypto.getDefaultParams helper, which will fill in any fields not supplied
+	 * with default values and validation the result.
 	 */
 	function CipherParams() {
 		this.algorithm = null;
 		this.keyLength = null;
 		this.mode = null;
 		this.key = null;
-		this.iv = null;
 	}
 	Crypto.CipherParams = CipherParams;
 
 	/**
-	 * Obtain a default CipherParams. This uses default algorithm, mode and
-	 * padding. If a key is specified this is used; otherwise a new key is generated
-	 * for the default key length. An IV is generated using the default
-	 * system SecureRandom.
-	 * A generated key may be obtained from the returned CipherParams
-	 * for out-of-band distribution to other clients.
-	 * @param key (optional) ArrayBuffer, Array, WordArray or base64 string, containing key
+	 * Obtain a complete CipherParams instance from the provided params, filling
+	 * in any not provided with default values, calculating a keyLength from
+	 * the supplied key, and validating the result.
+	 * @param params an object containing at a minimum a `key` key with value the
+	 * key, as either a binary (ArrayBuffer, Array, WordArray) or a
+	 * base64-encoded string. May optionally also contain: algorithm (defaults to
+	 * AES), mode (defaults to 'cbc')
 	 * @param callback (err, params)
 	 */
-	Crypto.getDefaultParams = function(key, callback) {
-		if(arguments.length == 1 && typeof(key) == 'function') {
-			callback = key;
-			key = undefined;
-		}
-		if(!key) {
-			generateRandom(DEFAULT_KEYLENGTH / 8, function(err, buf) {
-				if(err) {
-					callback(err);
-					return;
-				}
-				Crypto.getDefaultParams(buf, callback);
-			});
+	Crypto.getDefaultParams = function(params) {
+		var key;
+		/* Backward compatibility */
+		if((typeof(params) === 'function') || (typeof(params) === 'string')) {
+			Logger.deprecated('Crypto.getDefaultParams(key, callback)', 'Crypto.getDefaultParams({key: key})');
+			if(typeof(params) === 'function') {
+				Crypto.generateRandomKey(function(key) {
+					params(null, Crypto.getDefaultParams({key: key}));
+				})
+			} else {
+				callback = arguments[1];
```

Comment:
You should probably check that this second argument is in fact a callback before calling it, else we will simply get a runtime error in our library without any suitable error message for the dev.